selendroid-server/src/main/java/io/selendroid/server/action/ActionChain.java

```diff
@@ -18,10 +18,11 @@
   private int pauseTime = 0;
   private int index = 0;
 
-  public ActionChain(JSONObject actionChain) throws JSONException {
+  public ActionChain(JSONObject actionChain, int i) throws JSONException {
     this.inputDevice = actionChain.getString("inputDevice");
     this.actionChain = actionChain.getJSONArray("actions");
     this.context = new ActionContext();
+    this.context.setId(i);
```

Comment:
Each ActionChain needs a unique identifier to correspond with a pointer in the MotionEvent. When we implemented this we just set the id when creating the actionchain from Actions.java but only recently realized that the JSONObject representing an actionchain has a placeholder for 'id' which is set in MultitouchScreen in selendroid client. Would it be better to use that? It's currently unused and I believe this is its intended purpose but wasn't completely sure.